multiqc/modules/mirtop/mirtop.py

```diff
@@ -0,0 +1,242 @@
+        self.mirtop_data = self.ignore_samples(self.mirtop_data)
+
+        # Raise error if dict is empty
+        if len(self.mirtop_data) == 0:
+            raise UserWarning
+
+        log.info("Found {} reports".format(len(self.mirtop_data)))
+
+        # Write parsed report data to a file
+        self.write_data_file(self.mirtop_data, 'multiqc_mirtop')
+
+        # Create summary table
+        self.mirtop_stats_table()
+
+        # Create detailed plots
+        self.mirtop_barplot_section(aggregate_snps=True)
+
+    def parse_mirtop_report (self, f):
+        """ Parse the mirtop log file. """
+
+        content = json.loads(f['f'])
+        for s_name in content.get('metrics',{}).keys():
+            s_name = self.clean_s_name(s_name, f['root'])
+            ## Check for sample name duplicates
+            if s_name in self.mirtop_data:
+                log.debug("Duplicate sample name found! Overwriting: {}".format(s_name))
+            parsed_data = content['metrics'][s_name]
+            parsed_data['read_count'] = parsed_data['isomiR_sum'] + parsed_data['ref_miRNA_sum']
+            parsed_data['isomiR_perc'] = (parsed_data['isomiR_sum'] / parsed_data['read_count']) * 100
+            self.mirtop_data[s_name] = parsed_data
+
+    def mirtop_stats_table(self):
+        """ Take the parsed stats from the mirtop report and add them to the
+        basic stats table at the top of the report """
+
+        headers = OrderedDict()
+        headers['ref_miRNA_sum'] = {
+            'title': '{} Ref miRNA reads'.format(config.read_count_prefix),
+            'description': 'Read counts summed over all reference miRNAs ({})'.format(config.read_count_desc),
+            'modify': lambda x: x * config.read_count_multiplier,
+            'shared_key': 'read_count',
+            'scale': 'PuBu'
+        }
+        headers['isomiR_perc'] = {
+            'title': 'IsomiR %',
+            'description': '% of total read counts corresponding to isomiRs',
+            'min': 0,
+            'max': 100,
+            'suffix': '%',
+            'scale': 'YlOrRd'
+        }
+        headers['isomiR_sum'] = {
+            'title': '{} IsomiR reads'.format(config.read_count_prefix),
+            'description': 'Read counts summed over all isomiRs ({})'.format(config.read_count_desc),
+            'modify': lambda x: x * config.read_count_multiplier,
+            'shared_key': 'read_count',
+            'scale': 'Oranges'
+        }
+        headers['read_count'] = {
+            'title': '{} Reads'.format(config.read_count_prefix),
+            'description': 'Total read counts - both isomiRs and reference miRNA ({})'.format(config.read_count_desc),
+            'modify': lambda x: x * config.read_count_multiplier,
+            'shared_key': 'read_count',
+            'scale': 'BuGn'
+        }
+
+        self.general_stats_addcols(self.mirtop_data, headers)
+
+    def mirtop_barplot_section(self, aggregate_snps):
```

Comment:
Is there a reason to have `aggregate_snps` as a function argument? It seems that it is always `True`?

Reply:
I have removed this in my changes and just always apply this function now. Let me know if this is a mistake.

Reply:
Yes it's fine! I used it to test the output with/without the SNP aggregation, but it's easier to read with the aggregation.